data_safe_haven/external/api/graph_api.py

```diff
@@ -852,10 +1021,24 @@ def remove_user_from_group(
         try:
             user_id = self.get_id_from_username(username)
             group_id = self.get_id_from_groupname(group_name)
-            # Attempt to remove user from group
-            self.http_delete(
-                f"{self.base_endpoint}/groups/{group_id}/members/{user_id}/$ref",
-            )
+            # Check whether user is in group
+            json_response = self.http_get(
+                f"{self.base_endpoint}/groups/{group_id}/members",
+            ).json()
+            # Remove user from group if it is a member
+            if user_id in [
+                group_member["id"] for group_member in json_response["value"]
+            ]:
```

Comment:
```suggestion
            if user_id in (
                group_member["id"] for group_member in json_response["value"]
            ):
```

Got to save those clock cycles :smile:.